test/parallel/test-inspector-tracing-domain.js

```diff
@@ -52,7 +52,7 @@ async function test() {
                         'node.perf.timerify', 'v8'],
                        categories);
 
-  const traceConfig = { includedCategories: ['node'] };
+  const traceConfig = { includedCategories: ['v8'] };
```

Comment:
@eugeneo This test fails without this change. I suspect it has to do with the removal of the HandleWrap, right? Or is there something else at play? Would appreciate you having a look as the author of this test and someone with a lot of understanding of all this tracing stuff. Thanks!

Reply:
@nodejs/diagnostics Perhaps some of you might have insight too? Thanks!

> This test fails without this change. I suspect it has to do with the removal of the HandleWrap, right? Or is there something else at play? Would appreciate you having a look as the author of this test and someone with a lot of understanding of all this tracing stuff. Thanks!

Reply:
It's because timer_wrap.cc uses `AsyncWrap::MakeCallback()`, which calls `AsyncWrap::EmitTraceEventAfter()`.  Since you no longer use that method, it's not traced any more.

I personally don't feel too strongly but I guess it would be nice to keep the current behavior.

Reply:
I'd be okay with that.  In my opinion, AsyncWrap currently conflates too much (mushes too much into a single class) and could use some decoupling.

Reply:
LGTM on the test change. I haven't reviewed the rest of the PR.